### PR TITLE
[MM-57711] Convert ./components/latex_inline/latex_inline.tsx from Class Component to Function Component 

### PR DESCRIPTION
**Pull request by nikhilskul7**
Summary
Migrated latex_inline.tsx from Class Component to Function Component.

QA Test Steps:
Paste: e^{i\pi} + 1 = 0 wrapped by $ into the textarea. The latex inline component will render.

Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/26679
Jira https://mattermost.atlassian.net/browse/MM-57711

Release Note
```release-note
None
```

### CODE REVIEW

**Comment by nikhilskul7** [comment]
Hi @larkox,
I am trying to update the snapshot and tried to do it using this command `npm run test -- -u` but it did not work. ( It just ran tests but did not update the screenshot).
After this I went the the developer channel in mattermost and based on the suggestion I tried using `npm test -w channels -- -u latex_inline.test.tsx` but this just deleted the snapshot and did not update it.
Do you know how I can update the snapshot?

**Comment by larkox** [comment]
@nikhilskul7 have you tried running `npm run test -- -u` from the channels folder, instead of the webapp folder?

**Comment by nikhilskul7** [comment]
Thanks for the help @larkox .
Can you review my PR?

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
 
 import LatexInline from 'components/latex_inline/latex_inline';
 
-describe('components/LatexBlock', () => {
+import {withIntl} from 'tests/helpers/intl-test-helper';
+
+describe('components/LatexInline', () => {
     const defaultProps = {
-        content: 'e^{i\\pi} + 1 = 0',
+        content: '```latex e^{i\\pi} + 1 = 0```',
```
Comment (larkox) [inline webapp/channels/src/components/latex_inline/latex_inline.test.tsx:13]: This change shouldn't be needed, right?

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx [hunk 2]
```diff
@@ -24,19 +27,21 @@ describe('components/LatexBlock', () => {
...
             enableInlineLatex: true,
         };
 
-        const wrapper = shallow(<LatexInline {...props}/>);
-        await import('katex'); //manually import katex
-        expect(wrapper).toMatchSnapshot();
+        render(withIntl(<LatexInline {...props}/>));
+        const wrapper = await screen.findAllByTestId('latex-enabled');
+        expect(wrapper.length).toBe(1);
+        expect(wrapper.at(0)).toMatchSnapshot();
```
Comment (larkox) [inline webapp/channels/src/components/latex_inline/latex_inline.test.tsx:45]: `render` should return, among other things, the container. So you can do something like this:
```suggestion
        const {container} = render(withIntl(<LatexInline {...props}/>));
        expect(container).toMatchSnapshot();
```

**Review by larkox — CHANGES_REQUESTED** [review]
Looks great! Just a couple of minor things in the tests.

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx (hunk 2 above)
Comment (nikhilskul7) [reply webapp/channels/src/components/latex_inline/latex_inline.test.tsx:49]: Thanks for the suggestion, I have used this logic to update the test and also used 'act' to handle the state changes.

**Comment by nikhilskul7** [comment]
@larkox Thanks for the suggestions, I have updated my PR.
Can you review it again?
Also I noticed some tests are failing but I don't think the tests are failing due to my changes, are those flaky tests?  As I am seeing of those fail in master too.

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
         enableInlineLatex: true,
     };
 
     test('should match snapshot', async () => {
-        const wrapper = shallow(<LatexInline {...defaultProps}/>);
-        await import('katex'); //manually import katex
-        expect(wrapper).toMatchSnapshot();
+        let container;
+
+        await act(async () => {
```
Comment (larkox) [inline webapp/channels/src/components/latex_inline/latex_inline.test.tsx:21]: I am curious. Why `act` is needed?

**Review by larkox — APPROVED** [review]
Looks great. Thank you for the contribution!

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx (hunk 3 above)
Comment (nikhilskul7) [reply webapp/channels/src/components/latex_inline/latex_inline.test.tsx:21]: For the state change to take place correctly. Similar logic can be found in latex_block

Comment (larkox) [reply webapp/channels/src/components/latex_inline/latex_inline.test.tsx:21]: Makes sense. Thank you for the explanation!

**Comment by larkox** [comment]
@nikhilskul7 Yes, the `Enterprise CI/test` usually fail. Do not worry about them, we will get them sorted before QA review.

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx [hunk 4]
```diff
@@ -1,21 +1,28 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {shallow} from 'enzyme';
+import {render} from '@testing-library/react';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/latex_inline/latex_inline.test.tsx:4]: we should be using renderWithContext our own wrapper of testing-library

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx (hunk 3 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/latex_inline/latex_inline.test.tsx:21]: act is used internally in testing-library/react so we might not need this

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx (hunk 4 above)
Comment (nikhilskul7) [reply webapp/channels/src/components/latex_inline/latex_inline.test.tsx:4]: Done

**Comment by nikhilskul7** [comment]
Hi @M-ZubairAhmed ,
Can you take a look at this PR again?

File: webapp/channels/src/components/latex_inline/latex_inline.test.tsx (hunk 3 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/latex_inline/latex_inline.test.tsx:21]: `renderWithContext` uses `render` intenally which uses `act` internally, while `screen.findBy` are just selector function which doesnt wrap act.
However it seems that this behavior is due to async import of the module in `latex_inline.tsx` so here act is justified

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by larkox** [comment]
Thank you @nikhilskul7 for the contribution!

**Pull request merged**